### PR TITLE
Validate first and second data entries are equal

### LINKED ISSUE #633: Validate that first and second data entry are equal
Validate that first and second data entry are equal, otherwise move the data entry into a new status that should be resolved by the coordinator (part of epic #130).

### PR DESCRIPTION
**Pull request by praseodym**
Only create a polling station result when the first and second data entries are equal, otherwise return the new `FirstSecondEntryDifferent` status.

This new status is also added to the election status page (for coordinators).

~~This PR depends on changes from #740 (included as the first two commits), so draft until that PR is merged.~~

Resolves #633.

### CODE REVIEW

File: frontend/lib/ui/Badge/Badge.stories.tsx [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+];
+
 export const AllBadges: Story = () => {
   return (
     <>
-      <Badge type="definitive" />
-      <Badge type="not_started" />
-      <Badge type="first_entry_in_progress" showIcon />
+      {badgeTypes.map((type) => (
+        <Badge key={type} type={type} showIcon />
```
Comment (jschuurk-kr) [inline frontend/lib/ui/Badge/Badge.stories.tsx:22]: If we wrap this in a `<div id={type}>`, then we can get rid of the `.nth()`s in the e2e test. And we can expand the e2e test to check for `.toHaveAttribute('data-icon', 'IconPencil');` where relevant.

**Comment by oliver3** [comment]
You can still try and start data entry for a polling station that has differences, and will get an error message if you do so.

image

**Review by oliver3 — COMMENTED** [review]
I was not expecting the `finalise` call to return an HTTP Ok response if there were errors actually, but I don't have a better suggestion because returning an error while the finalised_at fields are changed is also not an option..

**Comment by praseodym** [comment]
> I was not expecting the `finalise` call to return an HTTP Ok response if there were differences actually, but I don't have a better suggestion because returning an error while the finalised_at fields are changed is also not an option..

For the typist (invoerder) everything is ok, the coordinator that needs to take action to resolve those differences (#130).

**Comment by Lionqueen94** [comment]
> You can still try and start data entry for a polling station that has differences, and will get an error message if you do so.

Fixed in 67247cfb4d8cef88970c77d30dc8b5b07a99c467

File: frontend/app/module/election/page/ElectionStatusPage.tsx [hunk 2]
```diff
@@ -70,6 +79,8 @@ function getTableHeaderForCategory(category: StatusCategory): ReactNode {
   );
 
   switch (category) {
+    case "errors_and_warnings":
+      return <CategoryHeader />;
     case "unfinished":
```
[the remark this answers is not in the document]
Comment (Lionqueen94) [reply frontend/app/module/election/page/ElectionStatusPage.tsx]: Improved in 67247cfb4d8cef88970c77d30dc8b5b07a99c467

**Review by oliver3 — APPROVED** [review]

**Review by jorisleker — CHANGES_REQUESTED** [review]
See last comment

**Comment by Lionqueen94** [comment]
> > > You can still try and start data entry for a polling station that has differences, and will get an error message if you do so.
> > 
> > 
> > Fixed in [67247cf](https://github.com/kiesraad/abacus/commit/67247cfb4d8cef88970c77d30dc8b5b07a99c467)
> 
> I can still trigger this error:
> 
>     * create two entries for a polling station that are different
> 
>     * on the pollingstationchoice page, enter the number of that polling station
> 
>     * click 'beginnen'
> 
>     * you will go to data entry and get this modal:
>       Scherm­afbeelding 2024-12-18 om 10 17 12
> 
>     * there is no way to close the modal, both buttons and the close icon will trigger it again
> 
> 
> Suggested fix: if the user presses 'beginnen', show the error we also show when the user tries to go to polling station that has finalized input: Scherm­afbeelding 2024-12-18 om 10 19 39
> 
> Bonus points: fix the layout bug in the "Stembureau 2 (Buurtcentrum de Mattenklopper) is al twee keer ingevoerd" feedback. This should be 1 sentence, instead of 3 inline blocks

Fixed in 105fac2aef1e4321d71003b8b3202314477f0d69. Please re-test :smile:

**Review by praseodym — COMMENTED** [review]
lgtm, but can't approve because I'm the PR author 🙃

**Review by jorisleker — APPROVED** [review]

**Review by cikzh — APPROVED** [review]
Reviewed by both me and @rnijveld

**Pull request merged**